test/mod_global_distrib_SUITE.erl

```diff
@@ -0,0 +1,175 @@
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+
+-module(mod_global_distrib_SUITE).
+-compile(export_all).
+-author('piotr.nosek@erlang-solutions.com').
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("exml/include/exml.hrl").
+
+%%--------------------------------------------------------------------
+%% Suite configuration
+%%--------------------------------------------------------------------
+
+all() ->
+    [{group, hook_handlers}].
+
+groups() ->
+    [
+     {hook_handlers, [], hook_handlers_tests()}
+    ].
+
+hook_handlers_tests() ->
+    [
+        missing_struct_in_message_from_user,
+        missing_struct_in_message_from_component
+    ].
+
+suite() ->
+    [].
+
+%%--------------------------------------------------------------------
+%% Init & teardown
+%%--------------------------------------------------------------------
+
+init_per_suite(Config) ->
+    {ok, _} = application:ensure_all_started(stringprep),
+    Config.
+
+end_per_suite(Config) ->
+    Config.
+
+init_per_group(_GroupName, Config) ->
+    Config.
+
+end_per_group(_GroupName, Config) ->
+    Config.
+
+init_per_testcase(_CaseName, Config) ->
+    set_meck(),
+    fake_start(),
+    Config.
+
+end_per_testcase(_CaseName, Config) ->
+    fake_stop(),
+    unset_meck(),
+    Config.
+
+%%--------------------------------------------------------------------
+%% Hook handlers tests
+%%--------------------------------------------------------------------
+
+
+%% missing_struct_ tests verify the behaviour of packet_to_component handler,
+%% which is supposed to update the mapping of the sender in Redis and cache.
+%% In case of routing between nodes in single cluster AND routers being reordered
+%% with component routers at the beginning of the chain, this hook must not fail
+%% despite lack of global_distrib structure in Acc.
+missing_struct_in_message_from_user(_Config) ->
+    From = jid:make(<<"user">>, global_host(), <<"resource">>),
+    {Acc, To} = fake_acc_to_component(From),
+    % The handler must not crash and return unchanged Acc
+    Acc = mod_global_distrib_mapping:packet_to_component(Acc, From, To).
+        
+%% Update logic has two separate paths: when a packet is sent by a user or by another
+%% component. This test covers the latter.
+missing_struct_in_message_from_component(_Config) ->
+    From = jid:make(<<"">>, <<"from_service.", (global_host())/binary>>, <<"">>),
+    {Acc, To} = fake_acc_to_component(From),
+    % The handler must not crash and return unchanged Acc
+    Acc = mod_global_distrib_mapping:packet_to_component(Acc, From, To).
```

Comment:
I am usually using `?assert_equal`.
Recently started to use 
`?assert_equal_extra(A, B, #{comment => "Do you have something to say?", some_var => X, some_other_var => Y})`
which contains enough info to figure out what is wrong, without reading logs or suite code in 50% of cases.